Sign in and sign up pages

jason1794c opened the pull request:
Created sign-in and sign-up pages with styling in the theme.js file. Modal sizing is not dynamic and is the same size for both the sign-in and sign-up pages. Please look at the code and let me know if there are any improvements or changes that I should make.

jason1794c committed:
Updated App.js

Included routes to sign-in and sign-up pages. User will be redirected to sign-in page if they are not signed in.

jason1794c committed:
Updated redirection

If user is not signed in, redirects to sign-in page. If user is signed in, redirects any attempts to sign-in or sign-up pages to the landing page.

jason1794c committed:
Create SignIn page

Created sign in page. Does not have close button at the moment.

jason1794c committed:
Create Sign Up page

Created sign up page. Checks for proper inputs such as email structure verification, and password length verification. Also checks for password confirmation. Does not have close button at the moment.

jason1794c committed:
Create Sign In styles

Styles for sign in page.

jason1794c committed:
Create sign up styles

Created styles for sign up page.

jason1794c committed:
Update pack-lock.json

jason1794c committed:
Create pack-lock.json

jason1794c committed:
Update package-lock.json

jason1794c committed:
Update Landing page

Temporary change to landing page.

jason1794c committed:
Update theme

Included styles to modal.

jason1794c committed:
Update App.js

Removed redirection to landing page

jason1794c committed:
Removed error message on input

Error text is removed when user begins inputting.

jason1794c committed:
Update App.js

Fixed ThemeProvider. Using Switch routing.

jason1794c committed:
Cleaned up spacing

Cleaned up spacing. Used styling from theme.js.

jason1794c committed:
Update theme.js

Added different Mui themes. Included styles for the sign-in and sign-up pages.

jason1794c committed:
Added comments for each style

Added comments.

jessdesr commented on client/src/pages/SignUp.js:
```diff
@@ -0,0 +1,247 @@
+            variant="outlined"
+            margin="normal"
+            fullWidth
+            name="passwordTwo"
```
maybe call this something like `passwordConfirm`

jessdesr commented on client/src/pages/SignUp.js:
```diff
@@ -0,0 +1,247 @@
+    errorText[field] = "";
+
+    // Check if we're entering in the password 2 field
+    if (field === "passwordTwo") {
```
I'd recommend using a form package like Formik to handle form validation, a lot cleaner than having to do it yourself

jessdesr reviewed (approved):
Good start, I think there are a lot of opportunities to clean things up (like with the styling), but that can come a bit later.  Make the change I recommend and then you can merge

jessdesr commented:
Also, in the future I'd like if your branch name matches the feature you're working on (in this case `fe-login-signup`), and to follow [this article](https://chris.beams.io/posts/git-commit) for info on making useful commit messages.

jason1794c committed:
Updated pack-lock.json

jason1794c committed:
Update package.json

Installed formik and formik-material-ui.

jason1794c committed:
Update the sign in button

Resized the sign in button to make it smaller using in-line styling.

jason1794c committed:
Replace validation code with formik validation

This change uses the formik component for validation. Material UI TextFields are replaced with formik form and fields in order to utilize formik's validation methods.

The onSubmit function requires code that will send a POST request of the new user's sign up details.

jason1794c committed:
Update formStyles and material UI's button style

Padding of material UI's button has been updated to 15px. The form styling has been changed to flexbox layout.

jason1794c committed:
Merge branch 'dev' into jason
